packages/core/echo/echo-pipeline/src/metadata/metadata-store.ts

```diff
@@ -81,6 +74,7 @@ export class MetadataStore {
       }
 
       this._metadata = schema.getCodecForType('dxos.echo.metadata.EchoMetadata').decode(data);
+      this._checkVersion();
```

Comment:
can you extract this higher up, (maybe ServiceContext?). This way metadata store can try loading old metadata for restore/backup purpoces.